### ISSUE TITLE
Setting defaultPaymentMethod in STPPaymentOptionsViewController has no effect

### ISSUE THREAD

**Issue by jose-t** [stripe/stripe-ios#1475]
## Summary
Setting `defaultPaymentMethod` from `STPPaymentOptionsViewController` seems to have no effect. The pre-selected payment method is always fetched from NSUserDefault.

## Code to reproduce
```
let paymentOptionsViewController = STPPaymentOptionsViewController(configuration: config, theme: STPTheme.default(), customerContext: customerContext, delegate: delegate)
paymentOptionsViewController.defaultPaymentMethod = defaultPaymentMethod
```

## iOS version
13.2

## Installation method
CocoaPods

## SDK version
18.3.0

## Other information
It works fine when I use `STPPaymentContext` to initialize the `STPPaymentOptionsViewController`.

**Comment by davidme-stripe** [comment]
Thanks for the report, we're working on a fix.

**Comment by yuki-stripe** [comment]
@jose-t This should be merged into master, and you can expect a release with this fix today!

I'm curious to learn more about your use case - why does the default behavior not work for you?

**Comment by jose-t** [comment]
It was fast, thanks you !

My application use the Stripe SDK on both iOS and Android. The user can add/remove cards and select his default payment method from the `STPPaymentOptionsController` and `PaymentMethodsActivity`.

Suppose the same user has two phones on iOS and Android, if he selects a default payment method on iOS, it will be registered in `NSUserDefaults`. Now he opens his Android phone and he select a new default payment method. If he reopens his iPhone and go back to the `STPPaymentOptionsController`, the pre-selected payment method will be the same stored in `NSUserDefaults` and not the default payment method stored on the stripe Customer object.

I think it could be dangerous to keep the default payment method state locally because it does not represent the real state.

**Comment by yuki-stripe** [comment]
Aha makes perfect sense, thank you!  This is helpful information as we plan future improvements here.

**Comment by davidme-stripe** [comment]
We released this in SDK v18.4.0.
